src/platforms/linux/daemon/dbusservice.cpp

```diff
@@ -130,11 +136,162 @@ QString DBusService::getLogs() {
 }
 
 bool DBusService::run(Op op, const Config& config) {
+  QStringList addresses;
+  for (const IPAddressRange& ip : config.m_allowedIPAddressRanges) {
+    addresses.append(ip.toString());
+  }
+
   return WgQuickProcess::run(
       op, config.m_privateKey, config.m_deviceIpv4Address,
       config.m_deviceIpv6Address, config.m_serverIpv4Gateway,
       config.m_serverIpv6Gateway, config.m_serverPublicKey,
       config.m_serverIpv4AddrIn, config.m_serverIpv6AddrIn,
-      config.m_allowedIPAddressRanges.join(", "), config.m_serverPort,
-      config.m_ipv6Enabled);
+      addresses.join(", "), config.m_serverPort, config.m_ipv6Enabled);
+}
+
+static inline bool parseEndpoint(struct sockaddr* endpoint,
```

Comment:
This method exists to verify that an endpoint is valid, correct.

1) Why not make this a general helper method that all platforms can use at a higher level?
2) Why do we need to do this at all - we are not trusting the endpoint that we get from mullvad api?

Reply:
It's not about verifying the endpoint, but this is about the creation of a `sockaddr`.
If you see, the WG API requires a `sockaddr` pointer in the `wg_peer` struct. This method "converts" a string (QString) + port into a `sockaddr`.

I don't know if we can make it cross-platform for the following reasons:
1. on windows `sockaddr` doesn't exist, and if it exists, is not what we should use. There is for sure a windows API...
2. the `wg_peer` exists only on linux. For mac and linux the communication works differently.

Let's see what we can "steal" from this implementation when working on macos.

Reply:
Thanks for the context. Perhaps the method name could hint more at this.